### PR TITLE
(PDB-1487) Add producer_timestamp to store-report

### PR DESCRIPTION
**Pull request by mullr**
The store-report command contains start_time and end_time, but those
are generated on the agent. This adds a producer_timestamp field to the
command, which is generated on the master. It can then be reliably
compared with the producer_timestamp on the deactivate-node command.

### CODE REVIEW

**Comment by ajroetker** [comment]
lgtm codewise

**Comment by wkalt** [comment]
does this require an update in the wire format doc?

**Comment by wkalt** [comment]
also in the reports api doc

**Comment by mullr** [comment]
updated api docs

File: src/puppetlabs/puppetdb/scf/migrate.clj [hunk 1]
```diff
@@ -1331,6 +1331,11 @@
         "ALTER TABLE fact_values RENAME COLUMN value_json TO value"
         "ALTER TABLE fact_values ALTER COLUMN value_json RENAME TO value"))))
 
+(defn add-producer-timestamp-to-reports []
+  (sql/do-commands
+   "ALTER TABLE reports ADD producer_timestamp TIMESTAMP WITH TIME ZONE"
+   "CREATE INDEX idx_reports_producer_timestamp ON reports(producer_timestamp)"))
+
```
Comment (rbrw) [inline src/puppetlabs/puppetdb/scf/migrate.clj]: @mullr wondered if just adding the column was sufficient.

**Comment by rbrw** [comment]
@mullr other than that, seems reasonable.

**Comment by ajroetker** [comment]
@mullr needs a rebase btw

**Comment by mullr** [comment]
Rebased, fixed migration to give `reports.producer_timestamp` an initial value from `end_time` and a `NOT NULL` constraint.

**Comment by senior** [comment]
@pljenkinsro retest this please

**Pull request merged**